tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -36,14 +38,26 @@ internal static class DatabaseMonitoringPropagator
 
         private static int _remainingErrorLogs = 100; // to prevent too many similar errors in the logs. We assume that after 100 logs, the incremental value of more logs is negligible.
 
-        internal static bool PropagateDataViaComment(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbCommand command, string configuredServiceName, string? dbName, string? outhost, Span span)
+        internal static bool PropagateDataViaComment(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbCommand command, string configuredServiceName, string? dbName, string? outhost, Span span, bool injectStoredProcedure)
         {
             if (integrationId is not (IntegrationId.MySql or IntegrationId.Npgsql or IntegrationId.SqlClient or IntegrationId.Oracle) ||
                 propagationLevel is not (DbmPropagationLevel.Service or DbmPropagationLevel.Full))
             {
                 return false;
             }
 
+            if (!injectStoredProcedure && command.CommandType == CommandType.StoredProcedure)
+            {
+                // we don't want to inject the comment for stored procedures if it has been disabled explicitly
+                return false;
+            }
+
+            if (command.CommandType == CommandType.StoredProcedure && integrationId != IntegrationId.SqlClient)
```

Comment:
I reckon we could use a single `if` statement here as we are repeating `command.CommandType == CommandType.StoredProcedure` twice otherwise. 

Did you choose to have two if statements for the sake of readability? 
I'm asking more because I'm curious more than anything🙂

Reply:
Thanks :)

No real reason I just didn't merge them, but merged into a single `if` now